superseriousbusiness/gotosocial
# [performance] media processing improvements

# Description
- pulls in latest `go-store` and `go-cache` versions
- wraps media encoding/decoding handling to be mostly contained within `gtsImage{}` and `gtsVideo{}` types for neatness
- decreases JPEG thumb quality to 70% (it's only a thumb, helps with filesize)
- uses memory pools for  JPEG and PNG encoding
- updates media processing to no longer use the atomic state variables (as these were mostly unused), all is now just done within mutex
- general tidy-up of media processing which improves: neatness, memory usage, error handling
- use larger write buffer size when streaming data into storage (now uses 16KiB, previously used 4KiB which is quite small)
- fix presigned S3 URL cache call (fixes #1279)
- changes test filenames / paths to fit with updated changes
- removes tee'd streaming of recache to both client and storage: in cases of slow client connections each tee stream could block a media worker pool function, which on instances with multiple users could very quickly exhaust available media workers
- run a disk storage clean operation (removes empty folders) on storage open

# Checklist
- [x] I/we have read the [GoToSocial contribution guidelines](https://github.com/superseriousbusiness/gotosocial/blob/main/CONTRIBUTING.md).
- [x] I/we have discussed the proposed changes already, either in an issue on the repository, or in the Matrix chat.
- [x] I/we have performed a self-review of added code.
- [x] I/we have written code that is legible and maintainable by others.
- [x] I/we have commented the added code, particularly in hard-to-understand areas.
- [x] I/we have made any necessary changes to documentation.
- [x] I/we have added tests that cover new code.
- [x] I/we have run tests and they pass locally with the changes.
- [x] I/we have run `go fmt ./...` and `golangci-lint run`.
